### PR TITLE
[PDI-17713] - GetPOP creates bad file names on duplicates

### PR DESCRIPTION
**Pull request by mbatchelor**

### CODE REVIEW

File: engine/src/main/java/org/pentaho/di/job/entries/getpop/MailConnection.java [hunk 1]
```diff
@@ -837,46 +825,55 @@ private void handlePart( String foldername, Part part, Pattern pattern ) throws
     }
   }
 
+  @VisibleForTesting
+  static String findValidTarget( String folderName, String fileName ) throws KettleException {
```
Comment (pentaho-lmartins) [inline engine/src/main/java/org/pentaho/di/job/entries/getpop/MailConnection.java]: Why not use `org.apache.commons.io.FilenameUtils` to get the filename without extension and extension.
The code would be cleaner.

Comment (mbatchelor) [reply engine/src/main/java/org/pentaho/di/job/entries/getpop/MailConnection.java]: Agreed - good call. I've updated the PR with your suggestion as well as improved the performance of the algorithm.

**Comment by mbatchelor** [comment]
Deleted old wingman report since I updated the PR.

**Comment by mbatchelor** [comment]
@pentaho-lmartins - Can you review my changes

**Comment by pentaho-lmartins** [comment]
Deleted Wingman reports, running Wingman again since this has been here for too long.

**Pull request merged**